cns/middlewares/k8sSwiftV2.go

```diff
@@ -135,150 +134,45 @@ func (m *K8sSWIFTv2Middleware) validateIPConfigsRequest(ctx context.Context, req
 	if err != nil {
-		return cns.PodIpInfo{}, errors.Wrapf(err, "failed to parse mtpnc primaryIP %s", mtpnc.Status.PrimaryIP)
+		return nil, errors.Wrapf(err, "failed to parse mtpnc primaryIP %s", mtpnc.Status.PrimaryIP)
 	}
 	// Get the IP address and prefix length
 	ip := p.Addr()
 	prefixSize := p.Bits()
 	if prefixSize != prefixLength {
-		return cns.PodIpInfo{}, errors.Wrapf(errInvalidMTPNCPrefixLength, "mtpnc primaryIP prefix length is %d", prefixSize)
+		return nil, errors.Wrapf(errInvalidMTPNCPrefixLength, "mtpnc primaryIP prefix length is %d", prefixSize)
 	}
-	podIPInfo := cns.PodIpInfo{
-		PodIPConfig: cns.IPSubnet{
-			IPAddress:    ip.String(),
-			PrefixLength: uint8(prefixSize),
-		},
-		MacAddress:        mtpnc.Status.MacAddress,
-		NICType:           cns.DelegatedVMNIC,
-		SkipDefaultRoutes: false,
-		// InterfaceName is empty for DelegatedVMNIC
-	}
-
-	return podIPInfo, nil
-}
 
-// setRoutes sets the routes for podIPInfo used in SWIFT V2 scenario.
-func (m *K8sSWIFTv2Middleware) setRoutes(podIPInfo *cns.PodIpInfo) error {
-	logger.Printf("[SWIFTv2Middleware] set routes for pod with nic type : %s", podIPInfo.NICType)
-	podIPInfo.Routes = []cns.Route{}
-	switch podIPInfo.NICType {
-	case cns.DelegatedVMNIC:
-		virtualGWRoute := cns.Route{
-			IPAddress: fmt.Sprintf("%s/%d", virtualGW, prefixLength),
-		}
-		// default route via SWIFT v2 interface
-		route := cns.Route{
-			IPAddress:        "0.0.0.0/0",
-			GatewayIPAddress: virtualGW,
-		}
-		podIPInfo.Routes = []cns.Route{virtualGWRoute, route}
-	case cns.InfraNIC:
-		// Get and parse infraVNETCIDRs from env
-		infraVNETCIDRs, err := configuration.InfraVNETCIDRs()
-		if err != nil {
-			return errors.Wrapf(err, "failed to get infraVNETCIDRs from env")
-		}
-		infraVNETCIDRsv4, infraVNETCIDRsv6, err := utils.ParseCIDRs(infraVNETCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse infraVNETCIDRs")
-		}
-
-		// Get and parse podCIDRs from env
-		podCIDRs, err := configuration.PodCIDRs()
-		if err != nil {
-			return errors.Wrapf(err, "failed to get podCIDRs from env")
-		}
-		podCIDRsV4, podCIDRv6, err := utils.ParseCIDRs(podCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse podCIDRs")
-		}
-
-		// Get and parse serviceCIDRs from env
-		serviceCIDRs, err := configuration.ServiceCIDRs()
-		if err != nil {
-			return errors.Wrapf(err, "failed to get serviceCIDRs from env")
-		}
-		serviceCIDRsV4, serviceCIDRsV6, err := utils.ParseCIDRs(serviceCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse serviceCIDRs")
-		}
-		// Check if the podIPInfo is IPv4 or IPv6
-		ip, err := netip.ParseAddr(podIPInfo.PodIPConfig.IPAddress)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse podIPConfig IP address %s", podIPInfo.PodIPConfig.IPAddress)
+	podIPInfos := make([]cns.PodIpInfo, len(mtpnc.Status.InterfaceInfos))
+	for i, interfaceInfo := range mtpnc.Status.InterfaceInfos {
```

Comment:
before reading this array, we still need the first podIpInfo to be from mtpnc.status.ncid, mtpnc.status.macaddress, etc the  upper fields.

Reply:
Addressed.